tests/k8st/utils/utils.py

```diff
@@ -19,6 +19,23 @@
 
 _log = logging.getLogger(__name__)
 
+# Helps with printing diags after a test.
+class DiagsCollector(object):
+    def __enter__(self):
+        pass
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        # Print out diagnostics for the test. These will go to screen
+        # on test failure.
+        _log.info("===================================================")
+        _log.info("============= COLLECTING DIAGS FOR TEST ===========")
+        _log.info("===================================================")
+        run("kubectl get deployments,pods,svc,endpoints --all-namespaces -o wide")
+        run("kubectl logs -n kube-system -l k8s-app=calico-node")
```

Comment:
While this command is nice and succinct, it looks like this will dump the logs in a fashion that's a little difficult to delimit them. 
Do you think we could execute something like `kubectl get pods -nkube-system | grep calico-node | awk '{ printf $1"\n" }'`, iterate over the result.split(), and print a header with the pod name + the log dump for each?

Reply:
It _should_ be easy enough to find the beginning of each - we include some standard logs on startup that should make it clear where one log file begins, and which node it is for.

Reply:
```
2018-12-06 21:51:20.987 [INFO][45] route_table.go 189: Queueing a resync of routing table. ipVersion=0x4
2018-12-06 21:51:20.989 [INFO][45] int_dataplane.go 765: Finished applying updates to dataplane. msecToApply=2.1740790000000003
2018-12-06 21:51:22.054 [INFO][45] health.go 150: Overall health summary=&health.HealthReport{Live:true, Ready:true}
2018-12-06 20:37:48.793 [INFO][8] startup.go 244: Early log level set to info
2018-12-06 20:37:48.793 [INFO][8] startup.go 264: Using HOSTNAME environment (lowercase) for node name
2018-12-06 20:37:48.794 [INFO][8] startup.go 272: Determined node name: kube-master
```
good enough for me :+1: